src/operator/tensor/cast_storage-inl.cuh

```diff
@@ -188,6 +188,15 @@ struct FillRspValsKernel {
   }
 };
 
+template<typename xpu, int ndim, typename DType>
+inline mshadow::Tensor<xpu, ndim, DType> AllocateTempDataForCast(const OpContext& op_ctx,
```

Comment:
1. I feel `GetTempSpace` is a better name.

2. Why is this function defined in .cuh but still using `xpu`? If it serves for general purpose, maybe we can put it somewhere accessible to all op files, such as "mxnet_op.h".

3. template function does not need to be "explicitly inlined".